app/src/main/java/com/example/habitsmasher/ui/dashboard/EditHabitFragment.java

```diff
@@ -112,17 +121,29 @@ public void onClick(View v) {
             }
         });
 
+        //set up days of the week buttons
+        setListenersForDaysOfTheWeek();
+
+        return view;
+    }
+
+    /*
+     * Sets up the buttons for the days of the week, including setting the initial states.
+     */
+    private void setListenersForDaysOfTheWeek(){
+
         //setting day initial states (clicked or not)
-        if (_tracker.getMonday()){mondayButton.performClick();}
-        if (_tracker.getTuesday()){tuesdayButton.performClick();}
-        if (_tracker.getWednesday()){wednesdayButton.performClick();}
-        if (_tracker.getThursday()){thursdayButton.performClick();}
-        if (_tracker.getFriday()){fridayButton.performClick();}
-        if (_tracker.getSaturday()){saturdayButton.performClick();}
-        if (_tracker.getSunday()){sundayButton.performClick();}
-
-        //day button onClick methods follow below
-        mondayButton.setOnClickListener(new View.OnClickListener() {
+        if (_tracker.getMonday()){_mondayButton.performClick();}
+        if (_tracker.getTuesday()){_tuesdayButton.performClick();}
+        if (_tracker.getWednesday()){_wednesdayButton.performClick();}
+        if (_tracker.getThursday()){_thursdayButton.performClick();}
+        if (_tracker.getFriday()){_fridayButton.performClick();}
+        if (_tracker.getSaturday()){_saturdayButton.performClick();}
+        if (_tracker.getSunday()){_sundayButton.performClick();}
```

Comment:
Minor: Related to above, you could also extract this code into its own method and keep this method named setListenersForDaysOfWeek()

Reply:
So, in onCreateView(), you can call this extracted method and then setListenersForDaysOfWeek().

Reply:
Sure thing, I'll do that in a bit.

Reply:
I'm gonna go with your suggestion above and simply rename the function to `initializeDaysOfTheWeekButtons()`